### PR TITLE
Closes #6282: Stop Firing WP Cron when disabled

### LINKED ISSUE #6282: Stop firing WP CRON when it's disabled
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version **Yes**
- Used the search feature to ensure that the bug hasn’t been reported before **Yes**

**Describe the bug**
When the constant `DISABLE_WP_CRON` is true, we need to stop calling the method spawn_cron here:

https://github.com/wp-media/wp-rocket/blob/06381b8860eb46e570911b2812652f54a676dd6d/inc/Engine/Optimization/RUCSS/Admin/Subscriber.php#L616

This was a suggestion from Kinsta not to fire the CRON because this may cause CPU issues.

**To Reproduce**
1. Add the following line in wp-config.php `define( 'DISABLE_WP_CRON', true )`
2. Activate RUCSS
3. Wait in WPR settings page for 90 seconds
4. Ajax request will be sent to our backend to fire the CRON

**Expected behavior**
CRON should not be called at all when this constant is set.

**Additional context**
Slack convos:

https://wp-media.slack.com/archives/CUT7FLHF1/p1700469134274159
https://wp-media.slack.com/archives/C056ZJMHG0P/p1700248478315299

**Acceptance Criteria (for WP Media team use only)**
1. When the CRON is disabled, we should run our functions manually to make sure that Used CSS for homepage is generated
2. We should not spawn CRON if it's disabled

### PR DESCRIPTION
**Pull request by Miraeld**
## Description

Add a check if CRON are disabled before firing one. 

Fixes #6282 

## Type of change
- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?
No

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [x] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

**Review by CrochetFeve0251 — APPROVED** [review]

**Comment by MathieuLamiot** [comment]
To keep track of our discussion @Miraeld, this could benefit from one or two test cases to make sure we don't create regressions later on on the use of DISABLE_WP_CRON

**Review by Mai-Saad — COMMENTED** [review]
@Miraeld Thanks for the PR. Shouldn't the condition [here](https://github.com/wp-media/wp-rocket/pull/6295/files#diff-9b19f89766a63e11ef01f00fc66257f99485141dd91781916529b2f14bb6bf56R354) be `typeof rocket_ajax_data.notice_end_time !== 'undefined' && ! rocket_ajax_data.cron_disabled` ?

**Comment by Miraeld** [comment]
> @Miraeld Thanks for the PR. Shouldn't the condition [here](https://github.com/wp-media/wp-rocket/pull/6295/files#diff-9b19f89766a63e11ef01f00fc66257f99485141dd91781916529b2f14bb6bf56R354) be `typeof rocket_ajax_data.notice_end_time !== 'undefined' && ! rocket_ajax_data.cron_disabled` ?

@Mai-Saad , You are right, I'm going to correct it.

**Review by mostafa-hisham — APPROVED** [review]

File: assets/js/wpr-admin.js [hunk 1]
```diff
@@ -351,7 +351,7 @@ if (typeof rocket_ajax_data.license_expiration !== 'undefined') {
   initializeClock('rocket-renew-countdown', rocket_ajax_data.license_expiration);
 }
 
-if (typeof rocket_ajax_data.notice_end_time !== 'undefined') {
+if (typeof rocket_ajax_data.notice_end_time !== 'undefined' && ! rocket_ajax_data.cron_disabled) {
```
Comment (wordpressfan) [inline assets/js/wpr-admin.js]: @Miraeld I believe we need to build the minified file.

**Review by wordpressfan — CHANGES_REQUESTED** [review]

File: assets/js/wpr-admin.js (hunk 1 above)
Comment (wordpressfan) [reply assets/js/wpr-admin.js]: @mostafa-hisham or it's built automatically using a github action?

Comment (wordpressfan) [reply assets/js/wpr-admin.js]: I generated it after changing the logic a bit.

**Review by CrochetFeve0251 — APPROVED** [review]

**Review by Mai-Saad — APPROVED** [review]
Working as expected.
Note: if cron is disabled, the success msg will mention home is processed while it isnot (will be handled in separate GH)
[testrail-report-549.pdf](https://github.com/wp-media/wp-rocket/files/13556304/testrail-report-549.pdf)

**Pull request merged**